openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -323,7 +323,9 @@ The following values are defined by this specification:
 * Grant Type `urn:ietf:params:oauth:grant-type:pre-authorized_code`:
   * `pre-authorized_code`: REQUIRED. The code representing the Credential Issuer's authorization for the Wallet to obtain Credentials of a certain type. This code MUST be short lived and single use. If the Wallet decides to use the Pre-Authorized Code Flow, this parameter value MUST be included in the subsequent Token Request with the Pre-Authorized Code Flow.
-  * `user_pin_required`: OPTIONAL. Boolean value specifying whether the AS expects presentation of a user PIN along with the Token Request in a Pre-Authorized Code Flow. Default is `false`. This PIN is intended to bind the Pre-Authorized Code to a certain transaction to prevent replay of this code by an attacker that, for example, scanned the QR code while standing behind the legitimate user. It is RECOMMENDED to send a PIN via a separate channel. If the Wallet decides to use the Pre-Authorized Code Flow, a PIN value MUST be sent in the `user_pin` parameter with the respective Token Request.
+  * `user_pin_required`: OPTIONAL. Boolean value specifying whether the AS expects presentation of a user PIN along with the Token Request in a Pre-Authorized Code Flow. Default is `false`. This PIN is intended to bind the Pre-Authorized Code to a certain transaction to prevent replay of this code by an attacker that, for example, scanned the QR code while standing behind the legitimate user. It is RECOMMENDED to send the PIN via a separate channel. If the Wallet decides to use the Pre-Authorized Code Flow, the PIN value MUST be sent in the `user_pin` parameter with the respective Token Request.
+  * `user_pin_length`: OPTIONAL. JSON Integer specifying the length of the PIN. This helps the Wallet to render the PIN screen and improve the user experience. When `user_pin_required` is `false`, this parameter MUST NOT be present.
+  * `user_pin_description`: OPTIONAL. JSON String containing guidance for the Holder of the Wallet how to obtain the PIN, e.g. specifying over which communication channel the PIN is delivered. The Wallet is RECOMMENDED to display this description next to the PIN entry form to improve the user experience. The length of the JSON String MUST be limited to 300 characters. The `user_pin_description` does not support internationalization (i18n), however the Issuer may detect the Holder's language by previous communication or an HTTP Accept-Language header within an HTTP GET request for a  `credential_offer_uri`. When `user_pin_required` is `false`, this parameter MUST NOT be present.
```

Comment:
```suggestion
  * `user_pin_length`: OPTIONAL. Integer specifying the length of the PIN. This helps the Wallet to render the PIN screen and improve the user experience. When `user_pin_required` is `false`, this parameter MUST NOT be present.
  * `user_pin_description`: OPTIONAL. String containing guidance for the Holder of the Wallet how to obtain the PIN, e.g., specifying over which communication channel the PIN is delivered. The Wallet is RECOMMENDED to display this description next to the PIN entry form to improve the user experience. The length of the JSON String MUST be limited to 300 characters. The `user_pin_description` does not support internationalization (i18n), however the Issuer may detect the Holder's language by previous communication or an HTTP Accept-Language header within an HTTP GET request for a  `credential_offer_uri`. When `user_pin_required` is `false`, this parameter MUST NOT be present.
```

Reply:
While I don't object to this, the specification is not very consistent on whether to name things "String" or "JSON String". Is there a consensus on this?

Reply:
My point of view is that it should be avoided to say "JSON String" (because the encoding is defined on the outer object and here we're just talking about the data structure).

Reply:
Note that the 'JSON String' and 'JSON Integer' language are used throughout OpenID Connect and https://datatracker.ietf.org/doc/html/rfc7519 etc.

We should probably agree on the policy and use that policy consistently rather than changing specific instances in unrelated pull requests.

Reply:
I'll open an issue!

Reply:
Since we agreed in #97 to drop "JSON", I have reopened this suggestion.

Reply:
applied